### PR TITLE
UIF-79 CRUD transfer - Fund options

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
Fix options list for fields `Transfer To` and `Transfer From`
https://issues.folio.org/browse/UIF-79

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/Budget/BudgetView/BudgetViewContainer.js [hunk 1]
```diff
@@ -117,6 +117,7 @@ const BudgetViewContainer = ({ history, resources }) => {
       </Pane>
       {isTransferModalOpened && (
         <TransferAddModal
+          budgetFund={budget.fundId}
```
Comment (aliaksei-chumakou) [inline src/components/Budget/BudgetView/BudgetViewContainer.js]: Why it's called `budgetFund`, but you pass there budget fund id?

File: src/Transactions/Transfer/TransferAddModalForm.js [hunk 2]
```diff
@@ -50,17 +93,21 @@ const TransferAddModalForm = ({ handleSubmit, onClose, funds }) => {
...
+              onChange={selectFromFund}
+              required={isTransferFromReqired}
+              {...validateTransferFrom}
             />
           </Col>
 
           <Col xs>
             <FieldSelect
-              dataOptions={funds}
+              dataOptions={getOptionsTo()}
```
Comment (aliaksei-chumakou) [inline src/Transactions/Transfer/TransferAddModalForm.js]: please don't do like this. You can define all required options above and pass here variables

File: src/Transactions/Transfer/TransferAddModalForm.js [hunk 3]
```diff
@@ -21,7 +28,43 @@ import {
...
 
-const TransferAddModalForm = ({ handleSubmit, onClose, funds }) => {
+const TransferAddModalForm = ({ budgetFund, handleSubmit, onClose, funds, store, dispatch, change }) => {
+  const formValues = getFormValues(TRANSFER_FORM)(store.getState());
+  const transferFrom = get(formValues, 'fromFundId');
+  const transferTo = get(formValues, 'toFundId');
+  const isTransferFromReqired = has(formValues, 'toFundId') ? transferTo !== budgetFund : false;
+  const validateTransferFrom = isTransferFromReqired ? { validate: validateRequired } : {};
+
+  const getOptionsFrom = () => (
```
Comment (aliaksei-chumakou) [inline src/Transactions/Transfer/TransferAddModalForm.js]: the same as above, what is the reason of defining a function here?

File: src/Transactions/Transfer/TransferAddModalForm.js [hunk 4]
```diff
@@ -21,7 +28,43 @@ import {
...
+    (!has(formValues, 'toFundId') ||
+    formValues.toFundId === budgetFund)
+      ? funds
+      : funds.filter(f => f.value === budgetFund)
+  );
+
+  const getOptionsTo = () => (
+    (!has(formValues, 'fromFundId') ||
```
Comment (aliaksei-chumakou) [inline src/Transactions/Transfer/TransferAddModalForm.js]: do you really need `has` here? IMHO, it's suitable for complex paths, but just for check key in object - it's overkill.
`if ('key' in formValues)`
`formValues.hasOwnProperty('key')`

File: src/Transactions/Transfer/TransferAddModalForm.js [hunk 5]
```diff
@@ -21,7 +28,43 @@ import {
 
 const TRANSFER_FORM = 'transferForm';
 
-const TransferAddModalForm = ({ handleSubmit, onClose, funds }) => {
+const TransferAddModalForm = ({ budgetFund, handleSubmit, onClose, funds, store, dispatch, change }) => {
+  const formValues = getFormValues(TRANSFER_FORM)(store.getState());
```
Comment (aliaksei-chumakou) [inline src/Transactions/Transfer/TransferAddModalForm.js]: usually we do `const formValues = getFormValues() || {}`

File: src/Transactions/Transfer/TransferAddModalForm.js [hunk 6]
```diff
@@ -21,7 +25,45 @@ import {
 
 const TRANSFER_FORM = 'transferForm';
 
-const TransferAddModalForm = ({ handleSubmit, onClose, funds }) => {
+const TransferAddModalForm = ({ fundId, handleSubmit, onClose, funds, store, dispatch, change }) => {
+  const formValues = getFormValues(TRANSFER_FORM)(store.getState()) || {};
+  const transferFrom = get(formValues, 'fromFundId');
+  const transferTo = get(formValues, 'toFundId');
+  const hasToFundIdProperty = Object.prototype.hasOwnProperty.call(formValues, 'toFundId');
```
Comment (aliaksei-chumakou) [inline src/Transactions/Transfer/TransferAddModalForm.js]: why not `formValues.hasOwnProperty('toFundId');` ?

Comment (AndreiShumski2) [reply src/Transactions/Transfer/TransferAddModalForm.js]: ESLint rule: Disallow use of Object.prototypes builtins directly (no-prototype-builtins)
https://eslint.org/docs/rules/no-prototype-builtins

File: src/Transactions/Transfer/TransferAddModalForm.js [hunk 7]
```diff
@@ -21,7 +25,45 @@ import {
 
 const TRANSFER_FORM = 'transferForm';
 
-const TransferAddModalForm = ({ handleSubmit, onClose, funds }) => {
+const TransferAddModalForm = ({ fundId, handleSubmit, onClose, funds, store, dispatch, change }) => {
+  const formValues = getFormValues(TRANSFER_FORM)(store.getState()) || {};
+  const transferFrom = get(formValues, 'fromFundId');
```
Comment (aliaksei-chumakou) [inline src/Transactions/Transfer/TransferAddModalForm.js]: you don't need `get` here

File: src/Transactions/Transfer/TransferAddModalForm.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/Transactions/Transfer/TransferAddModalForm.js]: fixed

File: src/Transactions/Transfer/TransferAddModalForm.js (hunk 3 above)
Comment (AndreiShumski2) [reply src/Transactions/Transfer/TransferAddModalForm.js]: fixed

File: src/Transactions/Transfer/TransferAddModalForm.js (hunk 4 above)
Comment (AndreiShumski2) [reply src/Transactions/Transfer/TransferAddModalForm.js]: done

File: src/Transactions/Transfer/TransferAddModalForm.js (hunk 6 above)
Comment (aliaksei-chumakou) [reply src/Transactions/Transfer/TransferAddModalForm.js]: so why not `'toFundId' in formValues` ?

Comment (AndreiShumski2) [reply src/Transactions/Transfer/TransferAddModalForm.js]: Done

File: src/Transactions/Transfer/TransferAddModalForm.js [hunk 8]
```diff
@@ -21,7 +24,45 @@ import {
...
+  );
+
+  const optionsTo = (
+    (!hasFromFundIdProperty ||
+    formValues.fromFundId === fundId)
+      ? funds
+      : funds.filter(f => f.value === fundId)
+  );
+
+  const selectFromFund = (e, id) => {
```
Comment (NikitaSedyx) [inline src/Transactions/Transfer/TransferAddModalForm.js]: please `useCallback`

**Pull request merged**